src/common/buffer_seastar.cc

```diff
@@ -24,9 +26,21 @@ namespace ceph::buffer {
 
 class raw_seastar_foreign_ptr : public raw {
   seastar::foreign_ptr<temporary_buffer> ptr;
+  seastar::alien::instance& alien;
  public:
   raw_seastar_foreign_ptr(temporary_buffer&& buf)
-    : raw(buf.get_write(), buf.size()), ptr(std::move(buf)) {}
+    : raw(buf.get_write(), buf.size()), ptr(std::move(buf)),
+      alien(seastar::engine().alien()) {}
+
+  ~raw_seastar_foreign_ptr() {
+    if (!seastar::engine_is_ready()) {
+      // we should let a seastar reactor destroy this memory, we are alien.
+      // alien::submit_to return type is std::future (not seastar::future)
+      seastar::alien::submit_to(alien, ptr.get_owner_shard(), [this]() {
+        return ptr.destroy();
+      }).wait();
```

Comment:
can we ignore the future here, wait() may cause performance issue.

Reply:
No, the documentation indicates that we are obligated to wait until the future resolves.

Reply:
@liu-chunmei, @athanatos - I think that we can capture the ptr by move and then it should be safe to not wait, PTAL.

Reply:
Might work

Reply:
Tests look ok, I'll merge the non-waiting version